libs/geometry/include/geometry/TangentSpaceMesh.h

```diff
@@ -0,0 +1,188 @@
+         */
+        DEFAULT = 0,
+
+        /*
+         * --- mikktspace ---
+         * Requires: normals + uvs + positions + indices
+         * Reference: Mikkelsen, M., 2008. Simulation of wrinkled surfaces revisited.
+         *            https://github.com/mmikk/MikkTSpace
+         *            https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes-overview
+         * Note: Will remesh
+         */
+        MIKKTSPACE = 1,
+
+        /*
+         * --- Lengyel's method ---
+         * Requires: normals + uvs + positions + indices
+         * Reference: Lengyel, E., 2019. Foundations of Game Engine Development: Rendering. Terathon
+         *     Software LLC.. (Chapter 7)
+         */
+        LENGYEL = 2,
+
+        /*
+         * --- Hughes-Moller method ---
+         * Requires: normals
+         * Reference: Möller, T. and Hughes, J.F., 1999. Efficiently building a matrix to rotate one
+         *     vector to another. Journal of graphics tools, 4(4), pp.1-4.
+         */
+        HUGHES_MOLLER =3,
+
+        /*
+         * --- Frisvad's method ---
+         * Requires: normals
+         * Reference: Frisvad, J.R., 2012. Building an orthonormal basis from a 3D unit vector
+         *     without normalization. Journal of Graphics Tools, 16(3), pp.151-159.
+         *     http://people.compute.dtu.dk/jerf/code/hairy/
+         */
+        FRISVAD = 4,
+
+        /*
+         * --- Flat Shading ---
+         * Requires: positions + indices
+         * Note: Will remesh
+         */
+        FLAT_SHADING = 5,
+
+        /*
+         * --- Sign of W ---
+         * Requires: normals + tangents
+         * Note: The sign of W determines the orientation of the bitangent.
+         */
+        SIGN_OF_W = 6
+    };
+
+    /*
+     * Computation of the tangent space is intended to be synchronous (worked on the same thread).
+     * Client is expected to keep the input immutable and in a good state for the duration of both
+     * computation *and* query. That is, when querying the result of the tangent spaces, part of the
+     * result might depend on the input data.
+     */
+    class Builder {
+    public:
+        Builder() noexcept;
+        ~Builder() noexcept;
+        Builder(Builder&& that) noexcept;
+        Builder& operator=(Builder&& that) noexcept;
+
+        Builder& vertexCount(size_t vertexCount) noexcept;
+
+        Builder& normals(const filament::math::float3*, size_t stride = 0) noexcept;
+        Builder& tangents(const filament::math::float4*, size_t stride = 0) noexcept;
+        Builder& uvs(const filament::math::float2*, size_t stride = 0) noexcept;
+        Builder& positions(const filament::math::float3*, size_t stride = 0) noexcept;
+
+        Builder& triangleCount(size_t triangleCount) noexcept;
+        Builder& triangles(const filament::math::uint3*) noexcept;
+        Builder& triangles(const filament::math::ushort3*) noexcept;
+
+        Builder& algorithmHint(AlgorithmHint) noexcept;
```

Comment:
by convention we always name parameters in headers. This will be needed when adding the doxygen doc.

Reply:
done